### ISSUE TITLE
Unable to access the API from GCP(Japan)

### ISSUE THREAD

**Issue by Azura-lee-code** [nobitex/docs-api#210]
### شرح خطا

Dear Nobitex team:

I am writing to seek assistance regarding a connectivity issue we are experiencing with the Nobitex API from our server hosted on Google Cloud. We are unable to access the API, and our attempts to connect are timing out.

Here is the command we are using:

```
curl --location 'https://api.nobitex.ir/market/stats' \
--header 'Content-Type: application/json' \
--data '{
      "srcCurrency": "USDT",
      "dstCurrency": "IRR"
}'
```
However, we are consistently encountering the following error:
`curl: (7) Failed to connect to api.nobitex.ir port 443: Connection timed out`

This issue seems to be specific to our server within Google Cloud, as we can access the API from local environment. We suspect there might be a networking or firewall issue that is preventing our Google Cloud server from accessing your API.

Could you please provide us with any insights or guidance on how to resolve this issue? Are there any known compatibility issues with Google Cloud, or are there specific network settings that we need to configure to enable access to your API?

Your prompt assistance in this matter would be greatly appreciated, as it is impacting our operational efficiency.

Thank you for your time and support.

### جزئیات درخواست

1. آدرس: `POST /sample/url`
2. داده:
    - key1: value1
    - key2: value2
    - ...
3. هدر:
    - Authorization: yes
    - ...
4. محیط:
    - تست‌نت
    - اصلی

### پاسخ دریافتی

skip

### پاسخ مورد انتظار

skip

**Comment by Azura-lee-code** [comment]
@rbabaee 
Thank you for your reply.
Here is response:
```
$ curl -D - -o /dev/null https://api.nobitex.ir/market/stats -vvv
* Expire in 0 ms for 6 (transfer 0x557504319110)
* Expire in 1 ms for 1 (transfer 0x557504319110)
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 0 ms for 1 (transfer 0x557504319110)
* Expire in 1 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 1 ms for 1 (transfer 0x557504319110)
* Expire in 1 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 1 ms for 1 (transfer 0x557504319110)
* Expire in 1 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 1 ms for 1 (transfer 0x557504319110)
* Expire in 1 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 2 ms for 1 (transfer 0x557504319110)
* Expire in 4 ms for 1 (transfer 0x557504319110)
* Expire in 3 ms for 1 (transfer 0x557504319110)
* Expire in 3 ms for 1 (transfer 0x557504319110)
* Expire in 4 ms for 1 (transfer 0x557504319110)
* Expire in 3 ms for 1 (transfer 0x557504319110)
* Expire in 3 ms for 1 (transfer 0x557504319110)
* Expire in 4 ms for 1 (transfer 0x557504319110)
* Expire in 4 ms for 1 (transfer 0x557504319110)
* Expire in 4 ms for 1 (transfer 0x557504319110)
* Expire in 4 ms for 1 (transfer 0x557504319110)
* Expire in 5 ms for 1 (transfer 0x557504319110)
* Expire in 5 ms for 1 (transfer 0x557504319110)
* Expire in 4 ms for 1 (transfer 0x557504319110)
* Expire in 5 ms for 1 (transfer 0x557504319110)
* Expire in 5 ms for 1 (transfer 0x557504319110)
* Expire in 8 ms for 1 (transfer 0x557504319110)
* Expire in 7 ms for 1 (transfer 0x557504319110)
* Expire in 7 ms for 1 (transfer 0x557504319110)
* Expire in 8 ms for 1 (transfer 0x557504319110)
* Expire in 7 ms for 1 (transfer 0x557504319110)
* Expire in 7 ms for 1 (transfer 0x557504319110)
* Expire in 8 ms for 1 (transfer 0x557504319110)
* Expire in 9 ms for 1 (transfer 0x557504319110)
* Expire in 9 ms for 1 (transfer 0x557504319110)
* Expire in 8 ms for 1 (transfer 0x557504319110)
* Expire in 9 ms for 1 (transfer 0x557504319110)
* Expire in 9 ms for 1 (transfer 0x557504319110)
* Expire in 16 ms for 1 (transfer 0x557504319110)
* Expire in 12 ms for 1 (transfer 0x557504319110)
* Expire in 12 ms for 1 (transfer 0x557504319110)
* Expire in 16 ms for 1 (transfer 0x557504319110)
* Expire in 13 ms for 1 (transfer 0x557504319110)
* Expire in 13 ms for 1 (transfer 0x557504319110)
* Expire in 16 ms for 1 (transfer 0x557504319110)
* Expire in 50 ms for 1 (transfer 0x557504319110)
* Expire in 50 ms for 1 (transfer 0x557504319110)
* Expire in 16 ms for 1 (transfer 0x557504319110)
* Expire in 50 ms for 1 (transfer 0x557504319110)
* Expire in 50 ms for 1 (transfer 0x557504319110)
* Expire in 32 ms for 1 (transfer 0x557504319110)
* Expire in 50 ms for 1 (transfer 0x557504319110)
* Expire in 50 ms for 1 (transfer 0x557504319110)
* Expire in 32 ms for 1 (transfer 0x557504319110)
* Expire in 50 ms for 1 (transfer 0x557504319110)
* Expire in 50 ms for 1 (transfer 0x557504319110)
* Expire in 64 ms for 1 (transfer 0x557504319110)
* Expire in 50 ms for 1 (transfer 0x557504319110)
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* Expire in 50 ms for 1 (transfer 0x557504319110)
* Expire in 64 ms for 1 (transfer 0x557504319110)
* Expire in 50 ms for 1 (transfer 0x557504319110)
* Expire in 50 ms for 1 (transfer 0x557504319110)
* Expire in 64 ms for 1 (transfer 0x557504319110)
* Expire in 50 ms for 1 (transfer 0x557504319110)
* Expire in 50 ms for 1 (transfer 0x557504319110)
* Expire in 64 ms for 1 (transfer 0x557504319110)
* Expire in 200 ms for 1 (transfer 0x557504319110)
* Expire in 200 ms for 1 (transfer 0x557504319110)
* Expire in 64 ms for 1 (transfer 0x557504319110)
* Expire in 200 ms for 1 (transfer 0x557504319110)
* Expire in 200 ms for 1 (transfer 0x557504319110)
* Expire in 128 ms for 1 (transfer 0x557504319110)
* Expire in 200 ms for 1 (transfer 0x557504319110)
* Expire in 200 ms for 1 (transfer 0x557504319110)
* Expire in 200 ms for 1 (transfer 0x557504319110)
*   Trying 185.143.233.120...
* TCP_NODELAY set
* Expire in 149672 ms for 3 (transfer 0x557504319110)
* Expire in 200 ms for 4 (transfer 0x557504319110)
  0     0    0     0    0     0      0      0 --:--:--  0:00:31 --:--:--     0* connect to 185.143.233.120 port 443 failed: Connection timed out
*   Trying 185.143.234.120...
* TCP_NODELAY set
* Expire in 133987 ms for 3 (transfer 0x557504319110)
  0     0    0     0    0     0      0      0 --:--:--  0:01:03 --:--:--     0* connect to 185.143.234.120 port 443 failed: Connection timed out
* Failed to connect to api.nobitex.ir port 443: Connection timed out
* Closing connection 0
curl: (7) Failed to connect to api.nobitex.ir port 443: Connection timed out
```

**Comment by amiraliakbari** [comment]
We have discussed the issue with our CDN provider and they said that the problem may be because of the source IP and some routing/reputation issues. Their proposed solution was to change the source IP address.

If changing IP is not possible for you or it does not solve the problem, we have a beta service that may be useful. You can try `https://api2.nobitex.ir` as the domain which is not using an external CDN, and check if it solves your problem. But be warned that it is still in beta phase and you should retest your code and have enough monitoring on it.

**Comment by Azura-lee-code** [comment]
@amiraliakbari 
Your assistance is greatly appreciated, but it appears that the beta URL did not resolve the issue. We received the following response.
```
$ curl -D - -o /dev/null https://api2.nobitex.ir/market/stats -vvv
* Expire in 0 ms for 6 (transfer 0x55b91d576110)
* Expire in 1 ms for 1 (transfer 0x55b91d576110)
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 0 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 1 ms for 1 (transfer 0x55b91d576110)
* Expire in 1 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 1 ms for 1 (transfer 0x55b91d576110)
* Expire in 1 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 1 ms for 1 (transfer 0x55b91d576110)
* Expire in 1 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 2 ms for 1 (transfer 0x55b91d576110)
* Expire in 4 ms for 1 (transfer 0x55b91d576110)
* Expire in 3 ms for 1 (transfer 0x55b91d576110)
* Expire in 3 ms for 1 (transfer 0x55b91d576110)
* Expire in 4 ms for 1 (transfer 0x55b91d576110)
* Expire in 3 ms for 1 (transfer 0x55b91d576110)
* Expire in 3 ms for 1 (transfer 0x55b91d576110)
* Expire in 4 ms for 1 (transfer 0x55b91d576110)
* Expire in 4 ms for 1 (transfer 0x55b91d576110)
* Expire in 4 ms for 1 (transfer 0x55b91d576110)
* Expire in 4 ms for 1 (transfer 0x55b91d576110)
* Expire in 5 ms for 1 (transfer 0x55b91d576110)
* Expire in 5 ms for 1 (transfer 0x55b91d576110)
* Expire in 4 ms for 1 (transfer 0x55b91d576110)
* Expire in 6 ms for 1 (transfer 0x55b91d576110)
* Expire in 6 ms for 1 (transfer 0x55b91d576110)
* Expire in 8 ms for 1 (transfer 0x55b91d576110)
* Expire in 8 ms for 1 (transfer 0x55b91d576110)
* Expire in 8 ms for 1 (transfer 0x55b91d576110)
* Expire in 8 ms for 1 (transfer 0x55b91d576110)
* Expire in 8 ms for 1 (transfer 0x55b91d576110)
* Expire in 8 ms for 1 (transfer 0x55b91d576110)
* Expire in 16 ms for 1 (transfer 0x55b91d576110)
* Expire in 11 ms for 1 (transfer 0x55b91d576110)
* Expire in 11 ms for 1 (transfer 0x55b91d576110)
* Expire in 16 ms for 1 (transfer 0x55b91d576110)
* Expire in 14 ms for 1 (transfer 0x55b91d576110)
* Expire in 14 ms for 1 (transfer 0x55b91d576110)
* Expire in 16 ms for 1 (transfer 0x55b91d576110)
* Expire in 15 ms for 1 (transfer 0x55b91d576110)
* Expire in 15 ms for 1 (transfer 0x55b91d576110)
* Expire in 32 ms for 1 (transfer 0x55b91d576110)
* Expire in 50 ms for 1 (transfer 0x55b91d576110)
* Expire in 50 ms for 1 (transfer 0x55b91d576110)
* Expire in 64 ms for 1 (transfer 0x55b91d576110)
* Expire in 50 ms for 1 (transfer 0x55b91d576110)
* Expire in 50 ms for 1 (transfer 0x55b91d576110)
* Expire in 64 ms for 1 (transfer 0x55b91d576110)
* Expire in 50 ms for 1 (transfer 0x55b91d576110)
* Expire in 50 ms for 1 (transfer 0x55b91d576110)
* Expire in 64 ms for 1 (transfer 0x55b91d576110)
* Expire in 50 ms for 1 (transfer 0x55b91d576110)
* Expire in 50 ms for 1 (transfer 0x55b91d576110)
* Expire in 64 ms for 1 (transfer 0x55b91d576110)
* Expire in 50 ms for 1 (transfer 0x55b91d576110)
* Expire in 50 ms for 1 (transfer 0x55b91d576110)
* Expire in 64 ms for 1 (transfer 0x55b91d576110)
* Expire in 50 ms for 1 (transfer 0x55b91d576110)
* Expire in 50 ms for 1 (transfer 0x55b91d576110)
* Expire in 64 ms for 1 (transfer 0x55b91d576110)
* Expire in 50 ms for 1 (transfer 0x55b91d576110)
* Expire in 50 ms for 1 (transfer 0x55b91d576110)
* Expire in 64 ms for 1 (transfer 0x55b91d576110)
* Expire in 200 ms for 1 (transfer 0x55b91d576110)
* Expire in 200 ms for 1 (transfer 0x55b91d576110)
* Expire in 64 ms for 1 (transfer 0x55b91d576110)
* Expire in 200 ms for 1 (transfer 0x55b91d576110)
* Expire in 200 ms for 1 (transfer 0x55b91d576110)
* Expire in 128 ms for 1 (transfer 0x55b91d576110)
* Expire in 200 ms for 1 (transfer 0x55b91d576110)
* Expire in 200 ms for 1 (transfer 0x55b91d576110)
* Expire in 200 ms for 1 (transfer 0x55b91d576110)
*   Trying 93.113.235.139...
* TCP_NODELAY set
* Expire in 200 ms for 4 (transfer 0x55b91d576110)
  0     0    0     0    0     0      0      0 --:--:--  0:00:31 --:--:--     0* connect to 93.113.235.139 port 443 failed: Connection timed out
* Failed to connect to api2.nobitex.ir port 443: Connection timed out
* Closing connection 0
curl: (7) Failed to connect to api2.nobitex.ir port 443: Connection timed out
```

**Comment by Azura-lee-code** [comment]
@amiraliakbari 
Thank you very much for your assistance and feedback. After conducting a thorough investigation, we have determined that the inability to use the API from the Iranian exchange is due to Google's restrictions on access in specific countries, including Iran. 
However, we are pleased to inform you that we have found a solution to address this issue and are taking steps to resolve it. Once again, we appreciate your assistance and your patience. Your support is greatly valued.
